src/Symfony/Component/DependencyInjection/Attribute/AutowireIterator.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Attribute;
+
+use Symfony\Component\DependencyInjection\Argument\IteratorArgument;
+use Symfony\Component\DependencyInjection\Argument\TaggedIteratorArgument;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\DependencyInjection\Exception\InvalidArgumentException;
+use Symfony\Component\DependencyInjection\TypedReference;
+use Symfony\Contracts\Service\Attribute\SubscribedService;
+use Symfony\Contracts\Service\ServiceSubscriberInterface;
+
+#[\Attribute(\Attribute::TARGET_PARAMETER)]
+class AutowireIterator extends Autowire
+{
+    /**
+     * @see ServiceSubscriberInterface::getSubscribedServices()
+     *
+     * @param array<string|SubscribedService>|string $services An explicit list of services or a tag name
+     */
+    public function __construct(
+        array|string $services,
+        string $indexAttribute = null,
+        string $defaultIndexMethod = null,
+        string $defaultPriorityMethod = null,
+        string|array $exclude = [],
```

Comment:
Is this an array of service ids? Also, super minor, we have `array|string` above and `string|array` here.

Reply:
Correct, see https://github.com/symfony/symfony/pull/44774/files#diff-e95ab96f0f1aa21244f00f2494014aae9e8d951e09c02e67c94e3a42bdbaba95R58
There's some overlap with `$excludeSelf` since the use case in the PR is to `$exclude = self::class`. We might have some legacy here but :shrug: